### ISSUE TITLE
Contributing and plans with this gem

### ISSUE THREAD

**Issue by g3ortega** [AirspaceTechnologies/google-cloud-vision#2]
Hi. I was about to create a wrapper for Cloud Vision API for a project I'm working, but I found this gem. I would like to contribute with you in case you have plans to improve it, maybe we can create a To Do, and you can share with me the plans you have with this gem.

**Comment by RyanRusnak** [comment]
I would be happy to work together on this! Since the API is still in beta, my immediate plans were just to keep up with their changes, but I would be happy to help you if you have something in mind. Want to start the todo list?

**Comment by g3ortega** [comment]
Great! 

Well, I was thinking in defining a public API that would be easy to use and keep the same as possible despite the future changes by Google. But, this is the difficult part, we have to think what would be the best way to interact with Cloud Vision API and bring a good experience for future users.

We can do a brainstorming about this, at the moment I have two ideas. 

Divide by type:

```ruby
GoogleCloudVision::LabelDetection.new [string or array of images], {options with defaults already defined}

GoogleCloudVision::LabelDetection.new '~/Pictures/dog.png', {max_results: 7}

=>  [{mid: "/m/068hy", description: "pet", score: 0.98293257}, {mid: "/m/0bt9lr", description: "dog", score: 0.98023009}, {mid: "/m/04rky", description: "mammal", score: 0.95344138}, {mid: "/m/0jbk", description: "animal", score: 0.92980206} ]

## We can also consider a way to pass an array that in return give us an array of images

GoogleCloudVision::LabelDetection.new ['~/Pictures/dog.png', '~/Pictures/cat.png'], {max_results: 8}

GoogleCloudVision::TextDetection.new './poster.jpg' # Using defaults 

```

With a focus on images, similar as you have it:
```ruby

GoogleCloudVision.new [string or array of images], {options with defaults already defined}

GoogleCloudVision.new '~/Pictures/poster.png', {detections: ['TEXT_DETECTION', 'LABEL_DETECTION'], max_results: 3 }

=> { label_annotations: [{mid: "/m/068hy", description: "food", score: =>0.98293257}, {mid: "/m/068hy", description: "menu", score: 0.88293257} ...], text_annotations: [{locale: "lt", description: "Antico Pizza Napoletana\n", bounding_poly: {vertices: [{x: 82, y: 22}, {x: 296, y: 22}, {x: 296, y: 84}, {x: 82, y: 84}]}]}

```
Another thing I would like is to be able to set up the a initial configuration file, where I can add the Cloud Vision API key, and modify some default values for options.

After that we can support URLs for images and not only local paths. And keep it updated with future changes and improvements of Cloud Vision API, you can count on me with those.

**Comment by RyanRusnak** [comment]
I think dividing by type is a greta idea as well as a config. I do not like having to pass the API key with each request. Want to open a PR and we can start implementing those changes?

**Comment by g3ortega** [comment]
Yup, tomorrow I'll be working on config, and then the other changes
